pages/resetpassword.tsx

```diff
@@ -0,0 +1,110 @@
+import { useRouter } from "next/router";
+import React, { useState } from "react";
+import { AuthContext } from "../context/AuthContext";
+import { getAuth, confirmPasswordReset } from "firebase/auth";
+import firebase from "firebase/compat/app";
+import styles from "../styles/ForgotPassword.module.css";
+import TextField from "@mui/material/TextField";
+
+const cssTextField = {
+    color: "black",
+    width: 690,
+    height: 80,
+    "& .MuiFilledInput-input": {
+      color: "#000000",
+    },
+    "& .MuiFormLabel-root": {
+      color: "#494C4E",
+    },
+};
+
+const ResetPassword: React.FC = () => {
+
+    const auth = getAuth();
+    const router = useRouter();
+
+    const [ newPassword, setNewPassword ] = useState<string>("");
+    const [ confirmNewPassword, setConfirmNewPassword ] = useState<string>("");
+    const [ error, setError ] = useState<string>("");
+
+    // firebase requires passwords to be greater than 6 characters
+    const passwordLengthOk = newPassword == "" || newPassword.length > 6;
+    const passwordsMatch = newPassword == "" || newPassword === confirmNewPassword;
+
+    const resetPassword = ():void => {
+
+        //retrieve code from url
+        const queryParams = new URLSearchParams(window.location.search);
+        const code = queryParams.get("oobCode")?.toString();
+
+        if (!passwordLengthOk){
+            setError("Password length must be atleast six.")
+
+        }
+        else if (!passwordsMatch){
+            setError("Passwords do not match.");
+        }
+        else if (code){
+            setError("");
+            firebase.auth().verifyPasswordResetCode(code)
+            .then(function(email) {
+                // Display a "new password" form with the user's email address
+            })
+            .catch(function() {
+                setError("Password could not be reset.");
+            })
+
+            firebase.auth().confirmPasswordReset(code, newPassword)
+            .then(function() {
+                router.push("/login");
+            })
+            .catch(function() {
+                setError("Password could not be reset.");
+            })
+        }
```

Comment:
this function should be in the AuthContext as well. Use the error field from there to handle any firebase errors

Also, can the errors be dynamically updated, instead of refreshing when you hit submit? The rest of the app does it that way so lets be consistent